### PR TITLE
refactor(check): update docs, allow for label to wrap input

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/1210

### CODE REVIEW

**Comment by srambach** [comment]
HBS parameters `check-label--type` and `check--type` are never used in any example or documented anywhere. If this is the mechanism to rearrange the html format for dropdown and select, I think how this is done needs to be well documented.

**Comment by mcoker** [comment]
@srambach would adding another demo showing the label wrapping the input and the span address your comment?

**Comment by srambach** [comment]
Yes, that would definitely help...maybe also some comments in the .hbs explaining the use of the parameter, since it isn't a standard one, and we don't have a corresponding "one doc to rule them all" to explain?

**Comment by mcoker** [comment]
fwiw, we use this format in other hbs files, too:

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Nav/nav-subnav-title.hbs
https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Nav/nav-section-title.hbs
https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/ModalBox/modal-box-header-title.hbs
https://github.com/christiemolloy/patternfly-next/blob/lists/src/patternfly/components/List/list.hbs
https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/InputGroup/input-group-text.hbs
https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/FormControl/form-control.hbs

**Comment by srambach** [comment]
OK, I hadn't seen it before.

**Comment by mcoker** [comment]
Added the example

**Comment by srambach** [comment]
Thanks. As a consumer, how would I know which version to use?

**Comment by mcoker** [comment]
I don't have a definitive answer to that :) As far as I know, I think a user can use whichever is appropriate. Though we could document in the select/dropdown multi-select examples why we're doing that (so that the whole list item is clickable) if you think that's valuable. Maybe @jgiardino has some guidance here.

**Comment by jgiardino** [comment]
I kinda feel like the rule should be: 
> When defining hover styles, these styles should be applied to the clickable element. 

And if further justification is needed:
> While the default styles applied to an element might not provide a visual indication of target area, the styles that display on hover should. To ensure that these styles accurately convey to the target area of an element where the user can click, hover styles should be applied to the clickable element of a component, and not to a larger wrapping element.

But even still, I think an additional statement might be needed depending on where we add this type of statement.

File: src/patternfly/components/Check/docs/code.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Checkbox and Radio are provided in the check component for use cases outside of forms. If they are used without label text ensure some sort of label for assistive technologies. (for example: `aria-label`)
 
+If you extend this component or modify the styles of this component, then make sure any hover styles defined are applied to the clickable elements, like `<input>` or `<label>` since hover styles are used to convey the clickable target area of an element. To maximize the target area, use the example html where the `<label>` is the wrapping element
+
```
Comment (jgiardino) [inline src/patternfly/components/Check/docs/code.md]: Thanks for adding this! Should we end with a period?

Comment (mcoker) [reply src/patternfly/components/Check/docs/code.md]: d'oh! added.

**Pull request merged**